### ISSUE TITLE
gdb autorepeat not working on plugin command

### ISSUE THREAD

**Issue by ThinerDAS** [hugsy/gef#419]
### Step 1: Describe your environment

  * Operating System: Fedora 29
  * Architecture: amd64
  * GDB version (including the Python library version): 

GDB Fedora 8.2-6.fc29 using Python engine 3.7

### Step 2: Describe your problem

Hitting Enter won't repeat the last command when a plugin command is involved, as far as I have tested.

#### Steps to reproduce

  1. Run any binary, interrupt
  2. `si`
  3. When the context comes back, hit Enter without command

#### Observed Results

  * What happened?  This could be a description, log output, etc.

It did nothing.

#### Expected results

  * What did you expect to happen?

It should execute `si` again.

#### Steps to reproduce 2

  1. Run any binary
  2. `vmmap`, assume that it is successful
  3. Hit Enter without command

#### Observed Results 2

  * What happened?  This could be a description, log output, etc.

It did nothing.

#### Expected results 2

  * What did you expect to happen?

It should execute `vmmap` again.

#### Comment

I tried to debug the problem.

I tested `p 1` and some small plugin, repeating works.

Working plugin example:
```
class TestCommand(gdb.Command):
    def __init__(self):
        super(TestCommand, self).__init__("testtest", gdb.COMMAND_USER)
    def invoke(self, arg, from_tty):
        print("Hello world")

TestCommand()
```

So the problem does not seem to be inside `gdb`.

When I commented out the `gdb.execute` line of hook_stop_handler, `ctx` won't be shown but `si` repeats successfully.

I stubbed `GenericCommand.invoke` to noop everything, but problem persists.

I hooked `GenericCommand.dont_repeat` but `ctx` did not invoke that function.

Python OOP is full of trap so I give up debugging for now. Any ideas of solving the problem?

**Comment by hugsy** [comment]
It is GDB, see #319

**Comment by ThinerDAS** [comment]
Yes, you are right.. And considering the number of times this issue is referenced ...

I monkey-patch my gdb into 8.2.1 and everything works as expected.

Fedora 29 / rawhide has not shipped gdb 8.2.1 yet, so I took archlinux version and copy its `/usr/bin/gdb` directly to the corresponding path, then made symbolic link for libreadline and libmpfr due to version mismatch.
